### PR TITLE
Add section "Backup and Restore" to the Cluster Administration Guide.

### PR DESCRIPTION
**Pull request by ahardin-rh**
This is a continuation of the work in https://github.com/openshift/openshift-docs/pull/2140

Addresses https://bugzilla.redhat.com/show_bug.cgi?id=1259544

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@adellape Can you please eyeball this before we merge? This is verified by QE. I updated all links to xrefs, cleared out commenting, and squashed. Thanks!

File: admin_guide/backup_restore.adoc [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+----
+# yum install etcd
+----
+
+Note the location of the *etcd* data directory (or `$ETCD_DATA_DIR` in the
+following sections), which depends on how *etcd* is deployed.
+
+[options="header",cols="1,2"]
+|===
+| Deployment | Data Directory
```
Comment (adellape) [inline admin_guide/backup_restore.adoc]: Maybe: s/Deployment/Deployment Type/

File: admin_guide/backup_restore.adoc [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+|all-in-one VM
```
Comment (adellape) [inline admin_guide/backup_restore.adoc]: Maybe: s/all-in-one VM/all-in-one cluster/

File: admin_guide/backup_restore.adoc [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+|*_/var/lib/openshift/openshift.local.etcd_*
+
+|external (not on master)
```
Comment (adellape) [inline admin_guide/backup_restore.adoc]: s/external/external etcd/

File: admin_guide/backup_restore.adoc [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+|*_/var/lib/etcd_*
+
+|embedded (on master)
```
Comment (adellape) [inline admin_guide/backup_restore.adoc]: s/embedded/embedded etcd/

File: admin_guide/backup_restore.adoc [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    --as-template=mycluster \//<2>
+    > mycluster.template.yaml
+----
+<1> Preserve fields that may be cluster specific,
+such as service `portalIP` values or generated names.
+<2> The output file has `kind: Template` and `metadata.name: mycluster`.
+====
++
+[IMPORTANT]
+====
```
Comment (adellape) [inline admin_guide/backup_restore.adoc]: This could probably come out of the IMPORTANT box? It's real big.

File: admin_guide/backup_restore.adoc [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+no-2	resourcequotas (aka 'quota')
+no-?	replicasets (aka 'rs')
+y	ReplicationController
+no-2	rolebindings
+y	Route
+no-3	secrets
+no-2	serviceaccounts
+y	Service
+no-2	users
+----
```
Comment (adellape) [inline admin_guide/backup_restore.adoc]: I'm not sure how to decipher the `no-#`s.... ?

**Comment by adellape** [comment]
@ahardin-rh Some comments inline, overall looks good tho.

File: admin_guide/backup_restore.adoc [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+<1> Preserve fields that may be cluster specific,
+such as service `portalIP` values or generated names.
+<2> The output file has `kind: Template` and `metadata.name: mycluster`.
+====
++
+[IMPORTANT]
+====
+The object types included in `oc export all` are:
+
```
Comment (ahardin-rh) [inline admin_guide/backup_restore.adoc]: @deads2k can you please take a look at this `oc export all` output? Is this correct? The formatting seems strange and I am not sure what was originally intended here. Thanks!

File: admin_guide/backup_restore.adoc (hunk 6 above)
Comment (ahardin-rh) [inline admin_guide/backup_restore.adoc]: @adellape yeah, not sure.
@deads2k any idea on the formatting here? Thanks!

Comment (deads2k) [inline admin_guide/backup_restore.adoc]: > @deads2k any idea on the formatting here? Thanks!

I don't recognize it the format, but the list looks correct (for now).  Seem like we might want to just define `all` and indicate that everything not listed isn't included.

**Comment by ahardin-rh** [comment]
@deads2k @adellape I made some updates. PTAL. Thanks!

File: admin_guide/backup_restore.adoc [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+<2> The output file has `kind: Template` and `metadata.name: mycluster`.
+====
++
+The object types included in `oc export all` are:
+
+====
+----
+BuildConfig
+Build
+componentstatuses ('cs')
```
Comment (deads2k) [inline admin_guide/backup_restore.adoc]: the "yes" and "no's" before were indicating which of these are included.  Some are and some aren't.  I just didn't know why they had numbers.

**Comment by ahardin-rh** [comment]
@deads2k Thanks for clarifying. This is updated.

File: admin_guide/backup_restore.adoc [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
++
+The object types included in `oc export all` are:
+
+====
+----
+BuildConfig
+Build
+DeploymentConfig
+ImageStream
+ImageStreamTag
```
Comment (deads2k) [inline admin_guide/backup_restore.adoc]: not gathered.

Comment (deads2k) [inline admin_guide/backup_restore.adoc]: > not gathered.

list looks good otherwise.

**Comment by ahardin-rh** [comment]
@deads2k Fixed. Thanks. Also took the list out of code block format and moved to a bulleted list.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../admin_guide/backup_restore.adoc#admin-guide-backup-and-restore[Backup and Restore]
|New topic discussing _back up_ (saving state to separate storage) and _restore_ (recreating state from separate storage) at the cluster level.
%

**Pull request merged**